lib/fog/compute/google/requests/reset_windows_password.rb

```diff
@@ -0,0 +1,151 @@
+        # @param zone [String] the name of the zone of the instance
+        # @param user [String] the user whose password should be reset
+        #
+        # @return [String] new password
+        #
+        # @see https://cloud.google.com/compute/docs/instances/windows/automate-pw-generation
+        def reset_windows_password(server:, user:)
+          instance = server.name
+          zone = server.zone
+          # Pull the e-mail address of user authenticated to API
+          email = @compute.request_options.authorization.issuer
+
+          # Create a new key
+          key = OpenSSL::PKey::RSA.new(2048)
+          modulus, exponent = get_modulus_exponent_in_base64(key)
+
+          # Get Old Metadata
+          old_metadata = server.metadata
+
+          # Create JSON Object with needed information
+          metadata_entry = get_json_string(user, modulus, exponent, email)
+
+          # Create new metadata object
+          new_metadata = update_windows_keys(old_metadata, metadata_entry)
+
+          # Set metadata on instance
+          server.set_metadata(new_metadata, false)
+
+          # Get encrypted password from Serial Port 4 Output
+          # TODO: Should use server.serial_port_output
+          # but unable to get port 4 right now
+
+          # If machine is booting for the first time, there appears to be a
+          # delay before the password appears on the serial port.
+          serial_port_output = get_server_serial_port_output(instance, zone, :port => 4).contents
+          while serial_port_output.empty?
+            sleep(5)
+            serial_port_output = get_server_serial_port_output(instance, zone, :port => 4).contents
+          end
+
+          # Parse and decrypt password
+          enc_password = get_encrypted_password_from_serial_port(serial_port_output, modulus)
+          password = decrypt_password(enc_password, key)
+
+          return password
+        end
+
+        def get_modulus_exponent_in_base64(key)
+          mod = [key.n.to_s(16)].pack("H*").strip
+          exp = [key.e.to_s(16)].pack("H*").strip
+          modulus = Base64.strict_encode64(mod).strip
+          exponent = Base64.strict_encode64(exp).strip
+          return modulus, exponent
+        end
+
+        def get_expiration_time_string
+          utc_now = Time.now.utc
+          expire_time = utc_now + 5 * 60
+          return expire_time.strftime("%Y-%m-%dT%H:%M:%SZ")
+        end
+
+        def get_json_string(user, modulus, exponent, email)
+          expire = get_expiration_time_string
+          data = { 'userName': user,
+                   'modulus': modulus,
+                   'exponent': exponent,
+                   'email': email,
+                   'expireOn': expire }
+          return ::JSON.dump(data)
+        end
+
+        def update_windows_keys(old_metadata, metadata_entry)
+          if old_metadata[:items]
+            new_metadata = Hash[old_metadata[:items].map { |item| [item[:key], item[:value]] }]
```

Comment:
Consider using `Hash#merge` here: https://ruby-doc.org/core-2.5.1/Hash.html#method-i-merge

Reply:
Perhaps I'm missing something, but here I am converting an Array of Hashes into a Hash.  
```
[{:key => "key1", :value => "value1"}, {:key => "key2", :value => "value2}]

{"key1" => "value1", "key2" => "value2"}
```
My understanding of merge is that it can be used to convert a hash into another hash.  Or perhaps there is a way to use it that I am not thinking of.

Reply:
Ooof, my bad. Nope, you're correct here then. Disregard this comment.